### PR TITLE
Add String.prototype.includes polyfill

### PR DESCRIPTION
**Pull request by cosmith**
Add a polyfill for `String.prototype.includes` taken from MDN (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/includes)
See https://github.com/facebook/react-native/issues/5727

### CODE REVIEW

**Comment by yungsters** [comment]
Would you mind making a couple nitty changes for this addition to comply with our style guides?

File: packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js [hunk 1]
```diff
@@ -83,3 +83,23 @@ if (!String.prototype.repeat) {
...
+  String.prototype.includes = function() {
+    'use strict';
+    if (typeof arguments[1] === "number") {
+      if (this.length < arguments[0].length + arguments[1].length) {
+        return false;
+      } else {
+        if (this.substr(arguments[1], arguments[0].length) === arguments[0]) {
+          return true;
+        }
+        else {
```
Comment (yungsters) [inline packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js]:
```
} else {
```

File: packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js [hunk 2]
```diff
@@ -83,3 +83,23 @@ if (!String.prototype.repeat) {
     return result;
   };
 }
+
+if (!String.prototype.includes) {
+  String.prototype.includes = function() {
+    'use strict';
+    if (typeof arguments[1] === "number") {
```
Comment (yungsters) [inline packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js]: Single quotes.

File: packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js (hunk 1 above)
Comment (yungsters) [inline packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js]: Actually, replace this entire block with:

```
return this.substr(...) === arguments[0];
```

**Comment by cosmith** [comment]
Sure thing!

**Comment by cosmith** [comment]
There's actually a bug in this implementation unless I'm going crazy... they're doing `arguments[1].length` when `typeof arguments[1] === 'number'`.

I'll update now with more readable parameter names.

**Comment by cosmith** [comment]
The sample code was completely wrong... This is the last time I copy/paste from MDN without testing.

Straight from the standard: http://www.ecma-international.org/ecma-262/6.0/#sec-string.prototype.includes

File: packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js [hunk 3]
```diff
@@ -83,3 +83,18 @@ if (!String.prototype.repeat) {
...
+if (!String.prototype.includes) {
+  String.prototype.includes = function(search, start) {
+    'use strict';
+    if (typeof start !== 'number') {
+      start = 0;
+    }
+    
+    if (start + search.length > this.length) {
+      return false;
+    } else {
```
Comment (vjeux) [inline packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js]: Nit you don't need the else since you have an early return above

**Comment by vjeux** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**